sound/soc/sof/topology.c

```diff
@@ -2523,23 +2540,23 @@ static int sof_widget_unload(struct snd_soc_component *scomp,
 			kfree(dai->dai_config);
 			list_del(&dai->list);
 		}
+
+		pipeline = snd_sof_pipeline_find(sdev, swidget->pipeline_id);
+
 		break;
 	case snd_soc_dapm_scheduler:
-
-		/* power down the pipeline schedule core */
 		pipeline = swidget->private;
-		ret = snd_sof_dsp_core_power_down(sdev, 1 << pipeline->core);
-		if (ret < 0)
-			dev_err(scomp->dev, "error: powering down pipeline schedule core %d\n",
-				pipeline->core);
-
-		/* update enabled cores mask */
-		sdev->enabled_cores_mask &= ~(1 << pipeline->core);
 
 		break;
 	default:
+		pipeline = snd_sof_pipeline_find(sdev, swidget->pipeline_id);
+
 		break;
 	}
+
+	if (pipeline)
+		sof_core_disable(sdev, pipeline->core);
```

Comment:
@lyakh widget_unload() would only ever be called during rmmod ie when the module is removed and not during a suspend. So it is unclear how this change helps your problem with suspend

Reply:
@ranj063 I did see the CI result change. But I'll do some testing on Monday, thanks for a heads up